### ISSUE TITLE
Streaming

### ISSUE THREAD

**Issue by adamralph** [adamralph/bau#124]
From https://jabbr.net/#/rooms/scriptcs in response to @nblumhardt

> - it's something I've been thinking about a lot, and there's been a twitter conversation about it today
> - the aim of Bau is to be completely target agnostic and to take the best bits of the current frameworks so I think it's something that would be very good for it to support
> - although having said that, the initial driver is to be able to build .net projects and at the moment I'm struggling to see how that really fits with streaming. after all, the heart of any build script is going to be call out to msbuild.exe which is a black box returning an error code
> - I did a comparison of the current Bau alpha against the gulp for building .net and here's what I came up with https://github.com/bau-build/bau/blob/dev/comparison/gulpfile.js
> - it's probably naive and can be improved (I've never use gulp before) so I'd love it if anyone could improve it
> - but I think I'm going to spike streaming support. it would be awesome to have it. sadly I don't see a way for me to dogfood it in Bau's own build script yet since that's just a series of callouts to msbuild.exe, xunit.console.exe and nuget.exe (like most .net projects)
> - aside from the streaming, of the various syntaxes, Bau is closest to gulp so far
> - comparisons to grunt, rake and gulp https://github.com/bau-build/bau/tree/dev/comparison

**Comment by nblumhardt** [comment]
> the heart of any build script is going to be call out to msbuild.exe which is a black box returning an error code

Maybe not necessarily - the goal is often to call CSC.EXE (or lang-x equivalent); the patchy support for MSBuild on non-Windows platforms leaves an opportunity here IMHO :)

**Comment by adamralph** [comment]
I agree, but it seems like a huge thing to take on. Almost all current .NET dev is done in Visual Studio (99%), Xamarin Studio or Monodevelop which ultimately all dump a load of .sln and .csproj onto the filesystem. That would need replacing. I've already day dreamed about writing a VS add-in which swaps out the use of .sln and .csproj files for bau .csx files but a day dream is all it is right now :wink:. I guess the landscape is already starting to shift with project K though...
